src/zenml/models/v2/core/model_version.py

```diff
@@ -343,6 +343,7 @@ def to_model_class(
             version=self.name,
             was_created_in_this_run=was_created_in_this_run,
             suppress_class_validation_warnings=suppress_class_validation_warnings,
+            model_version_id=self.id,
         )
         mv.model_version_id = self.id
```

Comment:
Probably not necessary anymore?

Reply:
I don't think so, this one is there to keep track of the already fetched/created models in a Model class itself. Used to ensure that we don't create the second model version in the second step and so on. I will revalidate this once again though.

Reply:
Actually my comment was just to the line `mv.model_version_id = self.id` which seems to be passed in the `__init__` call above

Reply:
Indeed, thanks!